### PR TITLE
ORCH-487 Add support for downloading artifacts without repox

### PR DESCRIPTION
**Pull request by alain-kermis-sonarsource**
Currently, Orchestrator supports all types of literals to retrieve artifacts, such as

- DEV
- DOGFOOD
- LATEST_RELEASE
- The above three can contain [<version>], i.e. DEV[10.1]
- Hardcoded version i.e. 10.3

With disabling JFrog anonymous usage/authentication, other open source projects that are using Orchestrator can no longer download artifacts from JFrog - unless they specify credentials

The new approach is as follows:

- if `orchestrator.artifactory.url` remains empty, default `https://repox.jfrog.io/repox` is used
- Setting `orchestrator.artifactory.url` to anything else **will be assumed to be a maven repository**

So, for testing, setting `orchestrator.artifactory.url` to `https://repo1.maven.org/maven2`, I tested using a few literals and/or versions (for SQ artifacts):

**1) LATEST_RELEASE[10.0]** - Finds latest in the series

[2024.02.06 09:05:45][c.s.o.l.Artifactory][INFO ] Downloading https://repo1.maven.org/maven2/org/sonarsource/sonarqube/sonar-application/10.0.0.68432/sonar-application-10.0.0.68432.zip
[2024.02.06 09:05:57][c.s.o.l.Artifactory][INFO ] Found [org.sonarsource.sonarqube:sonar-application:10.0.0.68432:zip] at https://repo1.maven.org/maven2/org/sonarsource/sonarqube/sonar-application/10.0.0.68432/sonar-application-10.0.0.68432.zip
[2024.02.06 09:05:57][c.s.o.l.MavenLocator][INFO ] Found [org.sonarsource.sonarqube:sonar-application:10.0.0.68432:zip] at /home/alain.kermis/.sonar/orchestrator/cache/3948b47be5795326d3e945ae84ff1f73/sonar-application-10.0.0.68432.zip

**2) 10.0.0.68432** - Exact build number

[2024.02.06 09:08:40][c.s.o.l.Artifactory][INFO ] Downloading https://repo1.maven.org/maven2/org/sonarsource/sonarqube/sonar-application/10.0.0.68432/sonar-application-10.0.0.68432.zip
[2024.02.06 09:08:53][c.s.o.l.Artifactory][INFO ] Found [org.sonarsource.sonarqube:sonar-application:10.0.0.68432:zip] at https://repo1.maven.org/maven2/org/sonarsource/sonarqube/sonar-application/10.0.0.68432/sonar-application-10.0.0.68432.zip
[2024.02.06 09:08:54][c.s.o.l.MavenLocator][INFO ] Found [org.sonarsource.sonarqube:sonar-application:10.0.0.68432:zip] at /home/alain.kermis/.sonar/orchestrator/cache/3948b47be5795326d3e945ae84ff1f73/sonar-application-10.0.0.68432.zip

**3) DEV | DOGFOOD** - Unsupported

java.lang.IllegalStateException: Unsupported version alias for [org.sonarsource.sonarqube:sonar-application:DEV:zip]
java.lang.IllegalStateException: Unsupported version alias for [org.sonarsource.sonarqube:sonar-application:DOGFOOD:zip]

### CODE REVIEW

**Review by jacek-poreda-sonarsource — APPROVED** [review]
LGTM 👍 

I've tested against private and public repositories, all good, though be wary that for some reason spring mockMVC for few unit tests is throwing error, and you need to specify explicitly that you want to return json.

File: sonar-orchestrator/src/main/java/com/sonar/orchestrator/config/Configuration.java [hunk 1]
```diff
@@ -47,6 +47,7 @@
 
 public class Configuration {
   private static final String ENV_SHARED_DIR = "SONAR_IT_SOURCES";
+
```
Comment (matteo-mara-sonarsource) [inline sonar-orchestrator/src/main/java/com/sonar/orchestrator/config/Configuration.java:50]: Less than nitpick: Wondering if this new line was intentional or not.

File: README.md
Comment (matteo-mara-sonarsource) [inline README.md:1]: I am wondering if it should be highlighted in the readme that not all those aliases will work when using an instance that is not the Sonar official.

**Review by matteo-mara-sonarsource — APPROVED** [review]
LGTM.

**Pull request merged**